bbc/simorgh
# WebCore like component health in storybook

Resolves WSTEAM1-164 && WSTEAM1-179

**Overall change:**
Adding a webcore like documentation and component health in storybook.
**Code changes:**

- Changed documentation extension inside docs folder from .md to .mdx so that they can be displayed in storybook.
- Created a new component that displays component health related documentation into storybook. More info in the docs section of the health component [here](https://5d28eb3fe163f6002046d6fa-hhryymzopd.chromatic.com/?path=/docs/components-healthfactor--health-factor-with-complete-docs).
- Added storybook background addon. This allows to change the background colour of the story. I have created 2 default colours which are CPS and Optimo. This has been added to avoid the need to change the background in every story we create:

<img width="1353" alt="Screenshot 2022-10-27 at 15 51 57" src="https://user-images.githubusercontent.com/90621252/198323672-fe2586ef-43a7-4a69-8318-e6b6c636f72e.png">

Testing notes:
For both local storybook and deployed storybook:
- Check that the Health Factors component is correctly rendering.
- Check that documentation below the Health Factors component is rendering correctly.
- Check that documentation story are rendering correctly in storybook.
- Might be worth checking for level A browsers. 

How to visualise any documentation stories:
- Navigate to storybook (either local or deployed e.g: [storybook for this branch](https://5d28eb3fe163f6002046d6fa-ghdncmvswo.chromatic.com/?path=/story/do
- Navigate to any documentation's story
<img width="223" alt="Screenshot 2023-01-26 at 11 06 26" src="https://user-images.githubusercontent.com/90621252/214821090-9ed2cb60-8d43-4cd6-91a1-080330504c6b.png">
cs-a11y-testing-cross-device--page)) 

How to visualise any Health Factors component:
- Navigate to storybook (either local or deployed e.g: [storybook for this branch](https://5d28eb3fe163f6002046d6fa-ghdncmvswo.chromatic.com/?path=/story/docs-a11y-testing-cross-device--page)) 
- Navigate to any component's story:
<img width="228" alt="Screenshot 2023-01-26 at 10 53 50" src="https://user-images.githubusercontent.com/90621252/214820527-81bfe2a9-9517-4616-81fb-69dec9ca8b9d.png">

- Navigate to the Docs tab:
<img width="1579" alt="Screenshot 2023-01-26 at 10 54 13" src="https://user-images.githubusercontent.com/90621252/214820622-800d69b9-b589-45d5-96ad-ff0c5bea0674.png">

- The Health Factors component should be at the top of this page after the story title:

---

- [ ] (BBC contributors only) This PR follows the [repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

**Testing:**

- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false yarn test:e2e:interactive`)
